src/zookeeper/zookeeper_session.cpp

```diff
@@ -161,6 +166,11 @@ int zookeeper_session::attach(void *callback_owner, const state_callback &cb)
             zoo_sasl_params_t sasl_params = {0};
             sasl_params.service = dsn::security::FLAGS_zookeeper_kerberos_service_name;
             sasl_params.mechlist = "GSSAPI";
+            rpc_address addr;
+            CHECK(addr.from_string_ipv4(dsn::security::FLAGS_zookeeper_sasl_service_fqdn),
```

Comment:
Use `GetAddrInfo` to check if fqdn is valid ?

Reply:
It's better to improve `rpc_address::from_string_ipv4` which use `GetAddrInfo` replace to `gethostbyname`. I will push one new pr later.

Reply:
![image](https://github.com/apache/incubator-pegasus/assets/48315319/34fb9315-6909-4c3d-983f-8e90dc8e1dbc)

Reply:
Good idea to declare this function as static ! However, since it would return int, how would we decide if it is successful ?